packages/oauth/src/index.ts

```diff
@@ -134,26 +134,32 @@ export class InstallProvider {
    */
   public async authorize(source: InstallationQuery<boolean>): Promise<AuthorizeResult> {
     try {
+      // Note that `queryResult` may unexpectedly include null values for some properties.
+      // For example, MongoDB can often save properties as null for some reasons.
+      // Inside this method, we should alwayss check if a value is either undefined or null.
       let queryResult;
       if (source.isEnterpriseInstall) {
         queryResult = await this.installationStore.fetchInstallation(source as InstallationQuery<true>, this.logger);
       } else {
         queryResult = await this.installationStore.fetchInstallation(source as InstallationQuery<false>, this.logger);
       }
 
-      if (queryResult === undefined) {
+      if (queryResult === undefined || queryResult === null) {
         throw new Error('Failed fetching data from the Installation Store');
       }
 
       const authResult: AuthorizeResult = {};
-      authResult.userToken = queryResult.user.token;
 
-      if (queryResult.team !== undefined) {
+      if (queryResult.user) {
+        authResult.userToken = queryResult.user.token;
+      }
+
+      if (queryResult?.team?.id) {
```

Comment:
Changed from `queryResult.team !== undefined`. As it's fine to skip any falsy values such as 0 and an empty string here, we can check if a value is truthy.

Reply:
Tiny little thing: do we need the optional chaining `?` operator on the `queryResult` variable here? Since we check for null or undefined on `queryResult` a few lines before?

Reply:
Good point. will remove the `?`